clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ClusterController.groovy

```diff
@@ -254,10 +250,10 @@ class ClusterController {
         return expandServerGroup(sortedServerGroups.last())
       case TargetServerGroup.LARGEST:
         // Choose the server group with the most instances, falling back to newest in the case of a tie.
-        return expandServerGroups(sortedServerGroups).sort { lhs, rhs ->
+        return expandServerGroup(sortedServerGroups.sort { lhs, rhs ->
```

Comment:
I've been wanting a way to avoid having to pre-expand server groups in order to sort by instance count.

Upon inspection of the Amazon and Titus caching agents, we are explicitly serializing instance details into `attributes` (and not just relationships). A spot check of other `*ServerGroupCachingAgents` makes me think that others are doing something similar.

Although they aren't "real" instances (missing health amongst other things), it's sufficient for the `AmazonClusterProvider` to use when the `INSTANCES` relationship is not populated.

Is this crazy? What say you @cfieber?

This only kicks in when `ClusterProvider. supportsMinimalClusters()` is `true` (Amazon-only right now with Titus support out for PR).

Reply:
it seems reasonable to me - would be sort of nice if we had a generic ClusterProvider spec that asserted the expectations around supportsMinimalClusters but not sure what exactly that might look like